File: ERP.Frontend/src/app/interfaces/RegexPatterns.ts
export const RegexPatterns = {
safeText    : /^[\p{L}0-9\s,.'\\-]+$/u,
phone       : /^\+?[\d][\d\s]{6,18}[\d]$/,
email: /^[a-zA-Z0-9._%+\-]+@[a-zA-Z0-9.\-]+\.[a-zA-Z]{2,}$/,
alphaNumeric: /^[\p{L}0-9\s]+$/u,
categoryCode : /^[\p{L}0-9_\-]+$/u,
hexColor     : /^#[0-9a-fA-F]{6}$/,
subdomainSlug: /^[a-z0-9]([a-z0-9-]*[a-z0-9])?$/,
alpha: /^[\p{L}\s]+$/u,
login: /^[a-z0-9_-]+$/,
barCode: /^\d{8,13}$/,
integer: /^-?\d+$/,
decimal: /^-?\d+(\.\d+)?$/
}

File: ERPSystem/ERP.TenantService/Application/DTOs/RegexPatterns.cs
namespace ERP.TenantService.Application.DTOs;

public static class RegexPatterns
{
    public const string SafeText = @"^[\p{L}0-9\s,.'\-]+$";
    public const string Phone = @"^\+?[\d][\d\s]{6,18}[\d]$";
    public const string AlphaNumeric = @"^[A-Za-z0-9]+$";
    public const string CategoryCode = @"^[A-Za-z0-9_\-]+$";
    public const string SubdomainSlug = @"^[a-z0-9]([a-z0-9-]*[a-z0-9])?$";
}